From 9b282369523ce2aabcf95369a2ed3867efb5a4ac Mon Sep 17 00:00:00 2001
From: Daniel Fahey <daniel.fahey@pm.me>
Date: Wed, 29 Oct 2025 15:22:01 +0000
Subject: [PATCH] fix: qt6_generate_deploy_app_script deprecated keyword

FILENAME_VARIABLE was deprecated in Qt 6.5 in favour of OUTPUT_SCRIPT, and became breaking in Qt 6.10.
See: https://doc.qt.io/qt-6/whatsnew65.html#qt-core-module and https://doc.qt.io/qt-6/qt-generate-deploy-app-script.html
---
 internal/frontend/bridge-gui/bridge-gui/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/internal/frontend/bridge-gui/bridge-gui/CMakeLists.txt b/internal/frontend/bridge-gui/bridge-gui/CMakeLists.txt
index 93a9f4992f679ab0502e36d01b8f6107c3d95284..a0ad070ae370a121f0e746680e5f05b774f5d4ff 100644
--- a/internal/frontend/bridge-gui/bridge-gui/CMakeLists.txt
+++ b/internal/frontend/bridge-gui/bridge-gui/CMakeLists.txt
@@ -175,7 +175,7 @@ install(TARGETS bridge-gui
 
 qt_generate_deploy_app_script(
         TARGET bridge-gui
-        FILENAME_VARIABLE deploy_script
+        OUTPUT_SCRIPT deploy_script
         NO_UNSUPPORTED_PLATFORM_ERROR)
 
 if(UNIX AND NOT APPLE)
From 7823fe4904186253798fc633724988d43dd642e0 Mon Sep 17 00:00:00 2001
From: Daniel Fahey <daniel.fahey@pm.me>
Date: Wed, 29 Oct 2025 15:22:17 +0000
Subject: [PATCH] fix: make libQt6WaylandEglClientHwIntegration conditional for
 Qt 6.10

This library is no longer present in Qt 6.10, but is still required for Qt 6.9 and earlier. Conditionally include it based on Qt version.
---
 internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake b/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
index c55cf3caaa4d8ce5654bb26cb37a83ff6d81ca0f..654518d31f8d52ea2dc8abde87e8c0dee6f56eff 100644
--- a/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
+++ b/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
@@ -60,7 +60,10 @@ AppendQt6Lib("libicudata.so.73")
 AppendQt6Lib("libQt6XcbQpa.so.6")
 AppendQt6Lib("libQt6WaylandClient.so.6")
 AppendQt6Lib("libQt6WlShellIntegration.so.6")
-AppendQt6Lib("libQt6WaylandEglClientHwIntegration.so.6")
+# libQt6WaylandEglClientHwIntegration is no longer present in Qt 6.10
+if(Qt6_VERSION VERSION_LESS "6.10.0")
+    AppendQt6Lib("libQt6WaylandEglClientHwIntegration.so.6")
+endif()
 AppendQt6Lib("libQt6EglFSDeviceIntegration.so.6")
 AppendQt6Lib("libQt6EglFsKmsSupport.so.6")
 AppendQt6Lib("libQt6Sql.so.6")
